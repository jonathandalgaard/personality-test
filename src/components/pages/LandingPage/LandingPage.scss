.landing-page {
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;

  .quizzes {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-gap: 2rem;
    list-style: none;
    margin: 0;
    padding: 0;
  }

  .paper {
    max-width: var(--content-width);
    display: flex;
    flex-direction: column;
    gap: 1rem;
    text-align: center;

    .button {
      margin: auto;
    }
  }
}
