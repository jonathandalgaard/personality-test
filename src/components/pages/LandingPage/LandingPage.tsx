import { Link } from 'react-router-dom';
import Paper from 'components/shared/Paper/Paper';
import { useGetQuizzesQuery } from 'redux/api';
import Loader from 'components/shared/Loader/Loader';
import './LandingPage.scss';

function LandingPage() {
  const { data: quizzes } = useGetQuizzesQuery();

  return (
    <article className="landing-page">
      {quizzes ? (
        <ul className="quizzes">
          {quizzes.map((quiz) => (
            <li key={quiz.id}>
              <Paper>
                <h2>{quiz.title}</h2>
                <Link to={`/quiz/${quiz.id}`} className="button">Take the quiz</Link>
              </Paper>
            </li>
          ))}
        </ul>
      ) : <Loader />}
    </article>
  );
}

export default LandingPage;
