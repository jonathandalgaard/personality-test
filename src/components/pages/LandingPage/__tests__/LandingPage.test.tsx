import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from 'redux/store.ts';
import { BrowserRouter } from 'react-router-dom';
import LandingPage from 'components/pages/LandingPage/LandingPage';

jest.mock('redux/store', () => ({
  __esModule: true,
  store: {
    getState: () => null,
    subscribe: () => null,
    dispatch: () => null,
  },
}));

jest.mock('redux/api', () => ({
  __esModule: true,
  default: () => null,
  useGetQuizzesQuery: () => ({
    data: [
      { id: 1, title: 'First quiz' },
      { id: 2, title: 'Second quiz' },
    ],
  }),
}));

test('Renders App correctly', () => {
  const component = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <LandingPage />
      </BrowserRouter>
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
