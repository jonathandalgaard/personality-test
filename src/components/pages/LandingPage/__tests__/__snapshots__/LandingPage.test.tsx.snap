// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders App correctly 1`] = `
<article
  className="landing-page"
>
  <ul
    className="quizzes"
  >
    <li>
      <div
        className="paper"
      >
        <h1>
          First quiz
        </h1>
        <a
          className="button"
          href="/quiz/1"
          onClick={[Function]}
        >
          Take the quiz
        </a>
      </div>
    </li>
    <li>
      <div
        className="paper"
      >
        <h1>
          Second quiz
        </h1>
        <a
          className="button"
          href="/quiz/2"
          onClick={[Function]}
        >
          Take the quiz
        </a>
      </div>
    </li>
  </ul>
  <img
    alt=""
    className="background-image"
    src="/stars.jpg"
  />
</article>
`;
